### ISSUE TITLE
Fix FP S1264 ('prefer-while'): Ignore condition-less for loops

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#2614]
The community suggested that the rule S1264 [should not trigger on condition-less for](https://community.sonarsource.com/t/typescript-s1264-should-not-trigger-on-condition-less-for/42603). Looking at RSPEC-1264, the rule is already supposed to make exception to condition-less for loops. The rule implementation should be updated so that it no longer raises issues on:

```javascript
for (;;) {
  // ...
}
```

**Comment by yassin-kammoun-sonarsource** [comment]
Fixed in https://github.com/SonarSource/eslint-plugin-sonarjs/pull/248
